src/js/ProviderManager.js

```diff
@@ -0,0 +1,55 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+/*globals async, DirectoryLinksProvider, Links, PlacesProvider, gUserDatabase, gPinnedLinks, gBlockedLinks*/
+
+/**
+ * Singleton that handles link providers.
+ */
+
+ "use strict";
+
+const ProviderManager = {
+
+  _initialized: false,
+
+  get initialized() {
+    return this._initialized;
+  },
+
+  set initialized(initialized) {
+    this._initialized = initialized;
+  },
+
+  init: async(function*(requestURL) {
```

Comment:
Could we think about doing initialization of all global components in one place. I I had a stab at doing that in:
https://github.com/mozilla/remote-newtab/pull/147/files#diff-a27a3470be339546b28849e120d7f4ccL127 

Would really like to hear what you think would be the best approach. Right now, it's really messy to try to track down where all initialization happens, and the order. If we made it a rule that components cannot initialize other components, but can only initialize themselves, then we might be able to detangle some of the code. 

Basically, we could asynchronously initialize all components by standardizing that all .init() functions return a promise (even if they don't do any async work). That would allow us to chain them together in the right order, etc. all in one place (fingers crossed!).